### PR TITLE
feat(SPV-755): add contact views

### PR DESCRIPTION
**Pull request by pawellewandowski98**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

# Description

Added 3 new views:
- Contacts - for admin
- Contacts - for user
- New Contact - for user

### CODE REVIEW

File: src/pages/contacts.jsx
Comment (chris-4chain) [inline src/pages/contacts.jsx:1]: ### It's a general request 🙈 

Please check whether or not, your changes are properly formatted using the rules from the `.prettierrc` file.

File: src/components/admin-contacts.jsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+const AdminRejectContactButton = ({contactToReject, handleRejectContact}) => {
+    if(handleRejectContact) {
+        return <Button
+            onClick={() => handleRejectContact(contactToReject)}
+        >
+            Reject contact
+        </Button>
+    } else {
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply src/components/admin-contacts.jsx]: Done

File: src/components/admin-contacts.jsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+const AdminAcceptContactButton = ({contactToAccept, handleAcceptContact}) => {
+    if(handleAcceptContact) {
+        return <Button
+            onClick={() => handleAcceptContact(contactToAccept)}
+        >
+            Accept contact
+        </Button>
+    } else {
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply src/components/admin-contacts.jsx]: Done

File: src/components/admin-contacts.jsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+const AdminConfirmContactButton = ({contactToConfirm, handleConfirmContact}) => {
+    if(handleConfirmContact) {
+        return <Button
+            onClick={() => handleConfirmContact(contactToConfirm)}
+        >
+            Confirm contact
+        </Button>
+    } else {
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply src/components/admin-contacts.jsx]: Done

File: src/components/admin-contacts.jsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+export const AdminContactsList = ({items, refetch}) => {
+    const { spvWalletClient } = useUser();
+    // const [info, setInfo] = useState('');
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply src/components/admin-contacts.jsx]: Done

File: src/components/admin-contacts.jsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                    logger.info(`Contact ${contact.id} confirmed`);
+                    refetch();
+                    // setRefreshData(+new Date());
+                })
+                .catch((e) => {
+                    logger.error(e);
+                    alert("ERROR: Could not confirm contact "+e.message);
+                });
+        }
+    };
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply src/components/admin-contacts.jsx]: Done, rafactored all into one method

File: src/components/contacts.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+const RejectContactButton = ({contactToReject, handleRejectContact}) => {
+    if(handleRejectContact) {
+        return <Button
+            onClick={() => handleRejectContact(contactToReject)}
+        >
+            Reject contact
+        </Button>
+    } else {
+        return null;
+    }
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply src/components/contacts.js]: Done

File: src/components/contacts.js
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply src/components/contacts.js:1]: Done, merged admin and user contacts view

File: src/pages/contacts.jsx
Comment (pawellewandowski98) [reply src/pages/contacts.jsx:1]: Done

**Review by chris-4chain — APPROVED** [review]

**Comment by wregulski** [comment]
testing

**Pull request merged**